storage/database/dynamodb.go

```diff
@@ -0,0 +1,479 @@
+}
+
+func (dynamo *dynamoDB) NewIterator() Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) NewIteratorWithStart(start []byte) Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) NewIteratorWithPrefix(prefix []byte) Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) createBatchWriteWorker() {
+	failCount := 0
+	tableName := dynamo.config.TableName
+	batchWriteInput := &dynamodb.BatchWriteItemInput{
+		RequestItems: map[string][]*dynamodb.WriteRequest{},
+	}
+	dynamo.logger.Info("generate a dynamoDB batchWrite worker", "time", time.Now())
+
+	for {
+		select {
+		case <-dynamo.quitCh:
+			dynamo.logger.Info("close a dynamoDB batchWrite worker", "time", time.Now())
+			return
+		case batchInput := <-dynamo.writeCh:
+			batchWriteInput.RequestItems[tableName] = batchInput.items
+
+			BatchWriteItemOutput, err := dynamo.db.BatchWriteItem(batchWriteInput)
+			numUnprocessed := len(BatchWriteItemOutput.UnprocessedItems)
+			for err != nil || numUnprocessed != 0 {
+				if err != nil {
+					failCount++
+					dynamo.logger.Warn("dynamoDB failed to write batch items", "err", err, "failCnt", failCount)
+					if failCount > dynamoMaxRetry {
+						dynamo.logger.Error("dynamoDB failed many times. sleep a second and retry",
+							"failCnt", failCount)
+						time.Sleep(time.Second)
+					}
+				}
+
+				if numUnprocessed != 0 {
+					dynamo.logger.Debug("dynamoDB batchWrite remains unprocessedItem",
+						"numUnprocessedItem", numUnprocessed)
+					batchWriteInput.RequestItems[tableName] = BatchWriteItemOutput.UnprocessedItems[tableName]
+				}
+
+				BatchWriteItemOutput, err = dynamo.db.BatchWriteItem(batchWriteInput)
+				numUnprocessed = len(BatchWriteItemOutput.UnprocessedItems)
+			}
+
+			failCount = 0
+			batchInput.wg.Done()
+		}
+	}
+}
+
+func (dynamo *dynamoDB) NewBatch() Batch {
+	return &dynamoBatch{db: dynamo, tableName: dynamo.config.TableName, wg: &sync.WaitGroup{}}
+}
+
+type dynamoBatch struct {
+	db         *dynamoDB
+	tableName  string
+	batchItems []*dynamodb.WriteRequest
+	size       int
+	wg         *sync.WaitGroup
+}
+
+func (batch *dynamoBatch) Put(key, val []byte) error {
+	data := DynamoData{Key: key, Val: val}
+	dataSize := len(val)
+	if dataSize == 0 {
+		return nil
+	}
+
+	// If the size of the item is larger than the limit, it should be handled in different way
+	if dataSize > dynamoWriteSizeLimit {
+		batch.wg.Add(1)
+		go func() {
+			failCnt := 0
+			batch.db.logger.Debug("write large size data into fileDB")
+
+			_, err := batch.db.fdb.write(item{key: key, val: val})
+			for err != nil {
+				failCnt++
+				batch.db.logger.Error("cannot write an item into fileDB. check the status of s3", "err", err,
+					"numRetry", failCnt)
+				time.Sleep(time.Second)
+
+				batch.db.logger.Warn("retrying write an item into fileDB")
+				_, err = batch.db.fdb.write(item{key: key, val: val})
+			}
+			batch.wg.Done()
+		}()
+		data.Val = overSizedDataPrefix
+		dataSize = len(data.Val)
```

Comment:
If the size of the item is larger than the limit, do we write the data to both fdb and DyanmoDB?

Reply:
I think calling `batch.db.Put` would be enough here.

Reply:
@aidan-kwon Could you please check this comment?

Reply:
I misunderstood. Please ignore this comment.